File: Programmers/js/77885.js
//https://school.programmers.co.kr/learn/courses/30/lessons/77885

function solution(numbers) {
	let answer = [];
	for (let num of numbers) {
		let res = BigInt(num).toString(2).split("");
		res = ["0", ...res];
		let idx = res.length - 1;
		while (true) {
			if (res[idx] === "0") {
				res[idx] = "1";
				break;
			}
			idx -= 1;
		}
		idx += 1;
		while (idx < res.length) {
			if (res[idx] === "1") {
				res[idx] = "0";
				break;
			}
			idx += 1;
		}
		answer.push(parseInt(res.join(""), 2));
	}
	return answer;
}

solution([2, 7]);

File: Programmers/python/77885.py
def solution(numbers):
    answer = []
    for num in numbers:
        num = list("0"+str(bin(num)[2:]));
        idx = len(num) -1;
        while True:
            if num[idx] == "0" :
                num[idx] = "1";
                break;
            idx -= 1;
        idx += 1;
        while idx < len(num):
            if num[idx] == "1":
                num[idx] = "0";
                break;
            idx += 1;
        answer.append(int(''.join(num),2));
    return answer;